gusb/gusb-device.c

```diff
@@ -889,6 +889,47 @@ g_usb_device_get_string_descriptor (GUsbDevice  *device,
 	return g_strdup ((const gchar *)buf);
 }
 
+/**
+ * g_usb_device_get_string_descriptor_bytes:
+ * @desc_index: the index for the string descriptor to retrieve
+ * @langid: the language ID
+ * @error: a #GError, or %NULL
+ *
+ * Get a raw string descriptor from the device. The returned string should be freed
+ * with g_bytes_unref() when no longer needed.
+ *
+ * Return value: (transfer full): a possibly UTF-16 string, or NULL on error.
+ *
+ * Since: 0.3.6
+ **/
+GBytes *
+g_usb_device_get_string_descriptor_bytes (GUsbDevice  *device,
+					  guint8       desc_index,
+					  guint16      langid,
+					  GError     **error)
+{
+	gint rc;
+	unsigned char buf[128];
+
+	g_return_val_if_fail (G_USB_IS_DEVICE (device), NULL);
+	g_return_val_if_fail (error == NULL || *error == NULL, NULL);
```

Comment:
`error` should allowed to be `NULL`, no?

Reply:
yup, it's saying if error != NULL and *error != NULL then return with a critical warning.

Reply:
Ah, ok.